### PR TITLE
Disable more flight modes

### PR DESCRIPTION
**Pull request by amilcarlucas**

### CODE REVIEW

**Comment by rmackay9** [comment]
Thanks, merged!

**Comment by amilcarlucas** [comment]
Thanks Peter, I will try to address your comments tomorrow

File: ArduCopter/AP_Arming.cpp [hunk 1]
```diff
@@ -212,12 +212,14 @@ bool AP_Arming_Copter::parameter_checks(bool display_failure)
         }
 
         // acro balance parameter check
+#if MODE_ACRO_ENABLED == ENABLED
```
[the remark this answers is not in the document]
Comment (amilcarlucas) [reply ArduCopter/AP_Arming.cpp]: Addressed in a PR I'm preparing.

File: ArduCopter/Log.cpp [hunk 2]
```diff
@@ -537,7 +537,8 @@ void Copter::Log_Write_GuidedTarget(uint8_t target_type, const Vector3f& pos_tar
     DataFlash.WriteBlock(&pkt, sizeof(pkt));
 }
 
-// precision landing logging
+#if MODE_THROW_ENABLED == ENABLED
```
[the remark this answers is not in the document]
Comment (amilcarlucas) [reply ArduCopter/Log.cpp]: "Not having these symbols around has the advantage that the compiler will fail during the compilation process"

That is the reason I did it, to make sure the code was clean

File: ArduCopter/mode.cpp [hunk 3]
```diff
@@ -38,9 +38,11 @@ Copter::Mode *Copter::mode_from_mode_num(const uint8_t mode)
     Copter::Mode *ret = nullptr;
 
     switch (mode) {
+#if MODE_ACRO_ENABLED == ENABLED
         case ACRO:
             ret = &mode_acro;
             break;
+#endif
 
         case STABILIZE:
```
[the remark this answers is not in the document]
Comment (amilcarlucas) [reply ArduCopter/mode.cpp]: I added ACRO and THROW, so I focused on those, but you are correct a PR will follow.

**Comment by peterbarker** [comment]
On Mon, 19 Mar 2018, Amilcar Lucas wrote:

> "Not having these symbols around has the advantage that the compiler will
> fail during the compilation process"
> 
> That is the reason I did it, to make sure the code was clean

You understand my argument about keeping the code clean, though?  We do 
need to avoid going overboard with the #ifdef's!  (I know I didn't help 
with the initial run-through of getting rid of modes :-) )  This is why 
that code-size branch of mine attacks things the way tridge suggested - 
stubbing stuff out at the build-system level rather than with #ifdef's.

**Pull request merged**